dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboProtocol.java

```diff
@@ -265,6 +266,12 @@ Invoker<?> getInvoker(Channel channel, Invocation inv) throws RemotingException
         int port = channel.getLocalAddress().getPort();
         String path = (String) inv.getObjectAttachmentWithoutConvert(PATH_KEY);
 
+        // if the environment variable DUBBO_PORT_TO_REGISTRY is added
+        String dubboPortToRegistry = ConfigUtils.getSystemProperty(org.apache.dubbo.config.Constants.DUBBO_PORT_TO_REGISTRY);
+        if(StringUtils.isNotEmpty(dubboPortToRegistry)) {
+            port = Integer.valueOf(dubboPortToRegistry);
```

Comment:
This may break `DUBBO_PORT_TO_REGISTRY`'s design

Reply:
So, How to solve not found exporter in exporterMap by serviceKey?

dubbo/dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo
/DubboProtocol.java
![image](https://github.com/apache/dubbo/assets/11389933/7a31e73f-2ca4-4747-8fcf-bde7a7521b49)

Reply:
As for https://github.com/apache/dubbo/issues/12798, it is recommended to use `dubbo.protocol.port` insteand of `DUBBO_PORT_TO_REGISTRY`

Reply:
get it